sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -768,6 +769,25 @@ object JdbcUtils extends Logging {
   }
 
   /**
+   * Parses the user specified customDataFrameColumnTypes option value string, and returns
+   */
+  def parseUserSpecifiedColumnTypes(
+       schema: StructType,
+       columnTypes: String,
+       nameEquality: Resolver): StructType = {
+    val userSchema = CatalystSqlParser.parseTableSchema(columnTypes)
+    // This is resolved by names, only check the column names.
+    userSchema.fieldNames.foreach { col =>
+      schema.find(f => nameEquality(f.name, col)).getOrElse {
+        throw new AnalysisException(
+          s"${JDBCOptions.JDBC_CUSTOM_DATAFRAME_COLUMN_TYPES} option column $col not found in " +
+            s"schema ${schema.catalogString}")
+      }
+    }
+    userSchema
```

Comment:
What is your expected behaviors when users-specified schema does not include all the columns of the underlying table schema?